pkg/kubefed2/join.go

```diff
@@ -81,6 +82,8 @@ func (o *joinFederationOptions) Bind(flags *pflag.FlagSet) {
 		"Name of the secret where the cluster's credentials will be stored in the host cluster. This name should be a valid RFC 1035 label. If unspecified, defaults to a generated name containing the cluster name.")
 	flags.BoolVar(&o.addToRegistry, "add-to-registry", false,
 		"Add the cluster to the cluster registry that is aggregated with the kubernetes API server running in the host cluster context.")
+	flags.BoolVar(&o.limitedScope, "limited-scope", false,
```

Comment:
(no action required) This implies that the user will only ever want to federate resources into the `--federation-namespace` they specified when they originally joined the cluster. What if they want to add more namespaces? This would limit the ability to specify either all namespaces or just one, but not X number of namespaces.

I think we may need to eventually provide the ability to specify multiple namespaces initially upon join. Then if they later want to add or reduce the number of namespaces, we could eventually provide an option to do that or just have them unjoin/rejoin the cluster as an expedient, but that could be catastrophic to their existing workload.

In any case, this is good enough for now unless we want to consider using a flag to list multiple namespaces and limit it to one for now.

Reply:
An informer can be configured to target a single namespace or all namespaces, but not multiple namespaces.  Supporting multiple namespaces is possible, but would require refactoring the sync controller to abstract use of the informers and to create an informer per targeted namespace.  Because of this complexity, and the lack of users (let alone ones calling for the feature), I call YAGNI.  If a user wants to target multiple namespaces, they can deploy a controller manager per namespace (this is a common pattern for namespaced controllers).

Reply:
Yeah that gets complicated. YAGNI sgtm.